// @fluentui/react-overflow/lib-amd/overflowContext.js
define(["require", "exports", "@fluentui/react-context-selector"], function (require, exports, react_context_selector_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useOverflowContext = exports.OverflowContext = void 0;
    exports.OverflowContext = react_context_selector_1.createContext(undefined);
    var overflowContextDefaultValue = {
        itemVisibility: {},
        groupVisibility: {},
        hasOverflow: false,
        registerItem: function () { return function () { return null; }; },
        updateOverflow: function () { return null; },
        registerOverflowMenu: function () { return function () { return null; }; },
    };
    /**
     * @internal
     */
    var useOverflowContext = function (selector) {
        return react_context_selector_1.useContextSelector(exports.OverflowContext, function (ctx) {
            if (ctx === void 0) { ctx = overflowContextDefaultValue; }
            return selector(ctx);
        });
    };
    exports.useOverflowContext = useOverflowContext;
});
