// @fluentui/react-combobox/lib-amd/contexts/useComboboxContextValues.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useComboboxContextValues = void 0;
    function useComboboxContextValues(state) {
        var activeOption = state.activeOption, appearance = state.appearance, focusVisible = state.focusVisible, open = state.open, registerOption = state.registerOption, selectedOptions = state.selectedOptions, selectOption = state.selectOption, setActiveOption = state.setActiveOption, setOpen = state.setOpen, size = state.size;
        var combobox = {
            activeOption: activeOption,
            appearance: appearance,
            focusVisible: focusVisible,
            open: open,
            registerOption: registerOption,
            selectedOptions: selectedOptions,
            selectOption: selectOption,
            setActiveOption: setActiveOption,
            setOpen: setOpen,
            size: size,
        };
        return { combobox: combobox };
    }
    exports.useComboboxContextValues = useComboboxContextValues;
});
