// @fluentui/react-combobox/lib-amd/contexts/useListboxContextValues.js
define(["require", "exports", "@fluentui/react-context-selector", "./ComboboxContext"], function (require, exports, react_context_selector_1, ComboboxContext_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useListboxContextValues = void 0;
    function useListboxContextValues(state) {
        var hasComboboxContext = react_context_selector_1.useHasParentContext(ComboboxContext_1.ComboboxContext);
        var activeOption = state.activeOption, focusVisible = state.focusVisible, multiselect = state.multiselect, registerOption = state.registerOption, selectedOptions = state.selectedOptions, selectOption = state.selectOption, setActiveOption = state.setActiveOption;
        // get register/unregister functions from parent combobox context
        var comboboxRegisterOption = react_context_selector_1.useContextSelector(ComboboxContext_1.ComboboxContext, function (ctx) { return ctx.registerOption; });
        var registerOptionValue = hasComboboxContext ? comboboxRegisterOption : registerOption;
        var listbox = {
            activeOption: activeOption,
            focusVisible: focusVisible,
            multiselect: multiselect,
            registerOption: registerOptionValue,
            selectedOptions: selectedOptions,
            selectOption: selectOption,
            setActiveOption: setActiveOption,
        };
        return { listbox: listbox };
    }
    exports.useListboxContextValues = useListboxContextValues;
});
