// @fluentui/react-table/lib-amd/hooks/createColumn.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.createTableColumn = void 0;
    var defaultCompare = function () { return 0; };
    var defaultRenderCell = function () {
        if (process.env.NODE_ENV !== 'production') {
            // eslint-disable-next-line no-console
            console.warn('@fluentui/react-table: You are using the default column renderCell function that renders null');
        }
        return null;
    };
    var defaultRenderHeaderCell = function () {
        if (process.env.NODE_ENV !== 'production') {
            // eslint-disable-next-line no-console
            console.warn('@fluentui/react-table: You are using the default column renderHeaderCell function that renders null');
        }
        return null;
    };
    /**
     * Helper function to create column definition with defaults
     * @param options - column definition options
     * @returns - column definition with defaults
     */
    function createTableColumn(options) {
        var columnId = options.columnId, _a = options.renderCell, renderCell = _a === void 0 ? defaultRenderCell : _a, _b = options.renderHeaderCell, renderHeaderCell = _b === void 0 ? defaultRenderHeaderCell : _b, _c = options.compare, compare = _c === void 0 ? defaultCompare : _c;
        return {
            columnId: columnId,
            renderCell: renderCell,
            renderHeaderCell: renderHeaderCell,
            compare: compare,
        };
    }
    exports.createTableColumn = createTableColumn;
});
