// @fluentui/react-table/lib-amd/hooks/useTableFeatures.js
define(["require", "exports", "react", "./useTableSelection", "./useTableSort", "./useTableColumnSizing"], function (require, exports, React, useTableSelection_1, useTableSort_1, useTableColumnSizing_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableFeatures = exports.defaultTableState = void 0;
    var defaultRowEnhancer = function (row) { return row; };
    exports.defaultTableState = {
        selection: useTableSelection_1.defaultTableSelectionState,
        sort: useTableSort_1.defaultTableSortState,
        getRows: function () { return []; },
        getRowId: function () { return ''; },
        items: [],
        columns: [],
        // eslint-disable-next-line @typescript-eslint/naming-convention
        columnSizing_unstable: useTableColumnSizing_1.defaultColumnSizingState,
        tableRef: React.createRef(),
    };
    function useTableFeatures(options, plugins) {
        if (plugins === void 0) { plugins = []; }
        var items = options.items, getRowId = options.getRowId, columns = options.columns;
        var getRows = function (rowEnhancer) {
            if (rowEnhancer === void 0) { rowEnhancer = defaultRowEnhancer; }
            return items.map(function (item, i) { var _a; return rowEnhancer({ item: item, rowId: (_a = getRowId === null || getRowId === void 0 ? void 0 : getRowId(item)) !== null && _a !== void 0 ? _a : i }); });
        };
        var initialState = {
            getRowId: getRowId,
            items: items,
            columns: columns,
            getRows: getRows,
            selection: useTableSelection_1.defaultTableSelectionState,
            sort: useTableSort_1.defaultTableSortState,
            // eslint-disable-next-line @typescript-eslint/naming-convention
            columnSizing_unstable: useTableColumnSizing_1.defaultColumnSizingState,
            tableRef: React.createRef(),
        };
        return plugins.reduce(function (state, plugin) { return plugin(state); }, initialState);
    }
    exports.useTableFeatures = useTableFeatures;
});
