// @fluentui/react-combobox/lib-amd/components/OptionGroup/useOptionGroupStyles.js
define(["require", "exports", "tslib", "@fluentui/react-theme", "@griffel/react"], function (require, exports, tslib_1, react_theme_1, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useOptionGroupStyles_unstable = exports.optionGroupClassNames = void 0;
    exports.optionGroupClassNames = {
        root: 'fui-OptionGroup',
        label: 'fui-OptionGroup__label',
    };
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: {
            display: 'flex',
            flexDirection: 'column',
            rowGap: react_theme_1.tokens.spacingHorizontalXXS,
            '&:not(:last-child)::after': tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ content: '""' }, react_1.shorthands.borderBottom(react_theme_1.tokens.strokeWidthThin, 'solid', react_theme_1.tokens.colorNeutralStroke2)), { display: 'block', paddingBottom: react_theme_1.tokens.spacingHorizontalXS }), react_1.shorthands.margin(0, "calc(" + react_theme_1.tokens.spacingHorizontalXS + " * -1)", react_theme_1.tokens.spacingVerticalXS)),
        },
        label: tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusMedium)), { color: react_theme_1.tokens.colorNeutralForeground3, display: 'block', fontSize: react_theme_1.tokens.fontSizeBase200, fontWeight: react_theme_1.tokens.fontWeightSemibold, lineHeight: react_theme_1.tokens.lineHeightBase200 }), react_1.shorthands.padding(react_theme_1.tokens.spacingHorizontalS, react_theme_1.tokens.spacingHorizontalSNudge)),
    });
    /**
     * Apply styling to the OptionGroup slots based on the state
     */
    var useOptionGroupStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.optionGroupClassNames.root, styles.root, state.root.className);
        if (state.label) {
            state.label.className = react_1.mergeClasses(exports.optionGroupClassNames.label, styles.label, state.label.className);
        }
        return state;
    };
    exports.useOptionGroupStyles_unstable = useOptionGroupStyles_unstable;
});
