// @fluentui/react-combobox/lib-amd/components/OptionGroup/useOptionGroup.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useOptionGroup_unstable = void 0;
    /**
     * Create the state required to render OptionGroup.
     *
     * The returned state can be modified with hooks such as useOptionGroupStyles_unstable,
     * before being passed to renderOptionGroup_unstable.
     *
     * @param props - props from this instance of OptionGroup
     * @param ref - reference to root HTMLElement of OptionGroup
     */
    var useOptionGroup_unstable = function (props, ref) {
        var labelId = react_utilities_1.useId('group-label');
        var label = props.label;
        return {
            components: {
                root: 'div',
                label: 'span',
            },
            root: react_utilities_1.getNativeElementProps('div', tslib_1.__assign({ ref: ref, role: 'group', 'aria-labelledby': label ? labelId : undefined }, props)),
            label: react_utilities_1.resolveShorthand(label, {
                defaultProps: {
                    id: labelId,
                    role: 'presentation',
                },
            }),
        };
    };
    exports.useOptionGroup_unstable = useOptionGroup_unstable;
});
