// @fluentui/react-text/lib-commonjs/components/presets/Caption1Stronger/Caption1Stronger.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Caption1Stronger = exports.caption1StrongerClassNames = void 0;
const react_1 = /*#__PURE__*/require("@griffel/react");
const createPreset_1 = /*#__PURE__*/require("../createPreset");
const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");
exports.caption1StrongerClassNames = {
  root: 'fui-Caption1Stronger'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/react_1.__styles({
  root: {
    Bahqtrf: "fk6fouc",
    Be2twd7: "fy9rknc",
    Bhrd7zp: "flh3ekv",
    Bg96gwp: "fwrc4pm"
  }
}, {
  d: [".fk6fouc{font-family:var(--fontFamilyBase);}", ".fy9rknc{font-size:var(--fontSizeBase200);}", ".flh3ekv{font-weight:var(--fontWeightBold);}", ".fwrc4pm{line-height:var(--lineHeightBase200);}"]
});
/**
 * Text preset component for the Caption1Stronger typography variant
 */
exports.Caption1Stronger = /*#__PURE__*/createPreset_1.createPreset({
  useStyles,
  className: exports.caption1StrongerClassNames.root,
  displayName: 'Caption1Stronger'
});
