// @fluentui/react-table/lib-commonjs/hooks/useTableFeatures.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTableFeatures = exports.defaultTableState = void 0;
const React = /*#__PURE__*/require("react");
const useTableSelection_1 = /*#__PURE__*/require("./useTableSelection");
const useTableSort_1 = /*#__PURE__*/require("./useTableSort");
const useTableColumnSizing_1 = /*#__PURE__*/require("./useTableColumnSizing");
const defaultRowEnhancer = row => row;
exports.defaultTableState = {
  selection: useTableSelection_1.defaultTableSelectionState,
  sort: useTableSort_1.defaultTableSortState,
  getRows: () => [],
  getRowId: () => '',
  items: [],
  columns: [],
  // eslint-disable-next-line @typescript-eslint/naming-convention
  columnSizing_unstable: useTableColumnSizing_1.defaultColumnSizingState,
  tableRef: /*#__PURE__*/React.createRef()
};
function useTableFeatures(options, plugins = []) {
  const {
    items,
    getRowId,
    columns
  } = options;
  const getRows = (rowEnhancer = defaultRowEnhancer) => items.map((item, i) => {
    var _a;
    return rowEnhancer({
      item,
      rowId: (_a = getRowId === null || getRowId === void 0 ? void 0 : getRowId(item)) !== null && _a !== void 0 ? _a : i
    });
  });
  const initialState = {
    getRowId,
    items,
    columns,
    getRows,
    selection: useTableSelection_1.defaultTableSelectionState,
    sort: useTableSort_1.defaultTableSortState,
    // eslint-disable-next-line @typescript-eslint/naming-convention
    columnSizing_unstable: useTableColumnSizing_1.defaultColumnSizingState,
    tableRef: React.createRef()
  };
  return plugins.reduce((state, plugin) => plugin(state), initialState);
}
exports.useTableFeatures = useTableFeatures;
