// @fluentui/react-field/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.makeDeprecatedField = exports.getDeprecatedFieldClassNames = exports.useField_unstable = exports.useFieldStyles_unstable = exports.renderField_unstable = exports.fieldClassNames = exports.Field = void 0;
var Field_1 = /*#__PURE__*/require("./Field");
Object.defineProperty(exports, "Field", {
  enumerable: true,
  get: function () {
    return Field_1.Field;
  }
});
Object.defineProperty(exports, "fieldClassNames", {
  enumerable: true,
  get: function () {
    return Field_1.fieldClassNames;
  }
});
Object.defineProperty(exports, "renderField_unstable", {
  enumerable: true,
  get: function () {
    return Field_1.renderField_unstable;
  }
});
Object.defineProperty(exports, "useFieldStyles_unstable", {
  enumerable: true,
  get: function () {
    return Field_1.useFieldStyles_unstable;
  }
});
Object.defineProperty(exports, "useField_unstable", {
  enumerable: true,
  get: function () {
    return Field_1.useField_unstable;
  }
});
// eslint-disable-next-line deprecation/deprecation
var makeDeprecatedField_1 = /*#__PURE__*/require("./util/makeDeprecatedField");
Object.defineProperty(exports, "getDeprecatedFieldClassNames", {
  enumerable: true,
  get: function () {
    return makeDeprecatedField_1.getDeprecatedFieldClassNames;
  }
});
Object.defineProperty(exports, "makeDeprecatedField", {
  enumerable: true,
  get: function () {
    return makeDeprecatedField_1.makeDeprecatedField;
  }
});
