// @fluentui/react-table/lib-amd/components/DataGridHeaderCell/useDataGridHeaderCellStyles.js
define(["require", "exports", "@griffel/react", "../TableHeaderCell/useTableHeaderCellStyles"], function (require, exports, react_1, useTableHeaderCellStyles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useDataGridHeaderCellStyles_unstable = exports.dataGridHeaderCellClassNames = void 0;
    exports.dataGridHeaderCellClassNames = {
        root: 'fui-DataGridHeaderCell',
        button: 'fui-DataGridHeaderCell__button',
        sortIcon: 'fui-DataGridHeaderCell__sortIcon',
        aside: 'fui-DataGridHeaderCell__aside',
    };
    /**
     * Apply styling to the DataGridHeaderCell slots based on the state
     */
    var useDataGridHeaderCellStyles_unstable = function (state) {
        useTableHeaderCellStyles_1.useTableHeaderCellStyles_unstable(state);
        state.root.className = react_1.mergeClasses(exports.dataGridHeaderCellClassNames.root, state.root.className);
        if (state.button) {
            state.button.className = react_1.mergeClasses(exports.dataGridHeaderCellClassNames.button, state.button.className);
        }
        if (state.sortIcon) {
            state.sortIcon.className = react_1.mergeClasses(exports.dataGridHeaderCellClassNames.sortIcon, state.sortIcon.className);
        }
        if (state.aside) {
            state.aside.className = react_1.mergeClasses(exports.dataGridHeaderCellClassNames.aside, state.aside.className);
        }
        return state;
    };
    exports.useDataGridHeaderCellStyles_unstable = useDataGridHeaderCellStyles_unstable;
});
