// @fluentui/react-text/lib-commonjs/components/presets/Body1Strong/Body1Strong.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Body1Strong = exports.body1StrongClassNames = void 0;
const react_1 = /*#__PURE__*/require("@griffel/react");
const createPreset_1 = /*#__PURE__*/require("../createPreset");
const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");
exports.body1StrongClassNames = {
  root: 'fui-Body1Strong'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/react_1.__styles({
  root: {
    Bahqtrf: "fk6fouc",
    Be2twd7: "fkhj508",
    Bhrd7zp: "fl43uef",
    Bg96gwp: "f1i3iumi"
  }
}, {
  d: [".fk6fouc{font-family:var(--fontFamilyBase);}", ".fkhj508{font-size:var(--fontSizeBase300);}", ".fl43uef{font-weight:var(--fontWeightSemibold);}", ".f1i3iumi{line-height:var(--lineHeightBase300);}"]
});
/**
 * Text preset component for the Body1Strong typography variant
 */
exports.Body1Strong = /*#__PURE__*/createPreset_1.createPreset({
  useStyles,
  className: exports.body1StrongClassNames.root,
  displayName: 'Body1Strong'
});
