// @fluentui/react-tabs/lib-amd/components/TabList/TabListContext.js
define(["require", "exports", "@fluentui/react-context-selector"], function (require, exports, react_context_selector_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTabListContext_unstable = exports.TabListProvider = exports.TabListContext = void 0;
    var tabListContextDefaultValue = {
        appearance: 'transparent',
        reserveSelectedTabSpace: true,
        disabled: false,
        selectedValue: undefined,
        onRegister: function () {
            /* noop */
        },
        onUnregister: function () {
            /* noop */
        },
        onSelect: function () {
            /* noop */
        },
        getRegisteredTabs: function () {
            return {
                registeredTabs: {},
            };
        },
        size: 'medium',
        vertical: false,
    };
    exports.TabListContext = react_context_selector_1.createContext(undefined);
    exports.TabListProvider = exports.TabListContext.Provider;
    var useTabListContext_unstable = function (selector) {
        return react_context_selector_1.useContextSelector(exports.TabListContext, function (ctx) {
            if (ctx === void 0) { ctx = tabListContextDefaultValue; }
            return selector(ctx);
        });
    };
    exports.useTabListContext_unstable = useTabListContext_unstable;
});
