// @fluentui/react-tabs/lib-amd/components/TabList/useTabListContextValues.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTabListContextValues_unstable = void 0;
    function useTabListContextValues_unstable(state) {
        var appearance = state.appearance, reserveSelectedTabSpace = state.reserveSelectedTabSpace, disabled = state.disabled, selectedKey = state.selectedValue, onRegister = state.onRegister, onUnregister = state.onUnregister, onSelect = state.onSelect, getRegisteredTabs = state.getRegisteredTabs, size = state.size, vertical = state.vertical;
        var tabList = {
            appearance: appearance,
            reserveSelectedTabSpace: reserveSelectedTabSpace,
            disabled: disabled,
            selectedValue: selectedKey,
            onSelect: onSelect,
            onRegister: onRegister,
            onUnregister: onUnregister,
            getRegisteredTabs: getRegisteredTabs,
            size: size,
            vertical: vertical,
        };
        return { tabList: tabList };
    }
    exports.useTabListContextValues_unstable = useTabListContextValues_unstable;
});
