// @fluentui/react-table/lib-amd/components/TableCellActions/useTableCellActionsStyles.js
define(["require", "exports", "@griffel/react"], function (require, exports, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableCellActionsStyles_unstable = exports.tableCellActionsClassNames = void 0;
    exports.tableCellActionsClassNames = {
        root: 'fui-TableCellActions',
    };
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: {
            backgroundColor: 'inherit',
            position: 'absolute',
            right: '0px',
            top: '50%',
            transform: 'translateY(-50%)',
            opacity: 0,
            marginLeft: 'auto',
        },
        visible: {
            opacity: 1,
        },
    });
    /**
     * Apply styling to the TableCellActions slots based on the state
     */
    var useTableCellActionsStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.tableCellActionsClassNames.root, styles.root, state.visible && styles.visible, state.root.className);
        return state;
    };
    exports.useTableCellActionsStyles_unstable = useTableCellActionsStyles_unstable;
});
