// @fluentui/react-table/lib-amd/components/Table/useTableStyles.js
define(["require", "exports", "@griffel/react", "@fluentui/react-theme"], function (require, exports, react_1, react_theme_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableStyles_unstable = exports.tableClassNames = exports.tableClassName = void 0;
    exports.tableClassName = 'fui-Table';
    exports.tableClassNames = {
        root: 'fui-Table',
    };
    var useTableLayoutStyles = react_1.makeStyles({
        root: {
            display: 'table',
            verticalAlign: 'middle',
            width: '100%',
            tableLayout: 'fixed',
        },
    });
    var useFlexLayoutStyles = react_1.makeStyles({
        root: {
            display: 'block',
        },
    });
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: {
            borderCollapse: 'collapse',
            backgroundColor: react_theme_1.tokens.colorSubtleBackground,
        },
    });
    /**
     * Apply styling to the Table slots based on the state
     */
    var useTableStyles_unstable = function (state) {
        var styles = useStyles();
        var layoutStyles = {
            table: useTableLayoutStyles(),
            flex: useFlexLayoutStyles(),
        };
        state.root.className = react_1.mergeClasses(exports.tableClassName, styles.root, state.noNativeElements ? layoutStyles.flex.root : layoutStyles.table.root, state.root.className);
        return state;
    };
    exports.useTableStyles_unstable = useTableStyles_unstable;
});
