// @fluentui/react-text/lib-commonjs/components/presets/Body1Stronger/Body1Stronger.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Body1Stronger = exports.body1StrongerClassNames = void 0;
const react_1 = /*#__PURE__*/require("@griffel/react");
const createPreset_1 = /*#__PURE__*/require("../createPreset");
const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");
exports.body1StrongerClassNames = {
  root: 'fui-Body1Stronger'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/react_1.__styles({
  root: {
    Bahqtrf: "fk6fouc",
    Be2twd7: "fkhj508",
    Bhrd7zp: "flh3ekv",
    Bg96gwp: "f1i3iumi"
  }
}, {
  d: [".fk6fouc{font-family:var(--fontFamilyBase);}", ".fkhj508{font-size:var(--fontSizeBase300);}", ".flh3ekv{font-weight:var(--fontWeightBold);}", ".f1i3iumi{line-height:var(--lineHeightBase300);}"]
});
/**
 * Text preset component for the Body1Stronger typography variant
 */
exports.Body1Stronger = /*#__PURE__*/createPreset_1.createPreset({
  useStyles,
  className: exports.body1StrongerClassNames.root,
  displayName: 'Body1Stronger'
});
