// @fluentui/react-badge/lib-amd/components/Badge/useBadgeStyles.js
define(["require", "exports", "tslib", "@griffel/react", "@fluentui/react-theme"], function (require, exports, tslib_1, react_1, react_theme_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useBadgeStyles_unstable = exports.badgeClassNames = void 0;
    exports.badgeClassNames = {
        root: 'fui-Badge',
        icon: 'fui-Badge__icon',
    };
    // The text content of the badge has additional horizontal padding, but there is no `text` slot to add that padding to.
    // Instead, add extra padding to the root, and a negative margin on the icon to "remove" the extra padding on the icon.
    var textPadding = react_theme_1.tokens.spacingHorizontalXXS;
    var useRootClassName = react_1.makeResetStyles(tslib_1.__assign(tslib_1.__assign({ display: 'inline-flex', boxSizing: 'border-box', alignItems: 'center', justifyContent: 'center', position: 'relative' }, react_theme_1.typographyStyles.caption1Strong), { height: '20px', minWidth: '20px', padding: "0 calc(" + react_theme_1.tokens.spacingHorizontalXS + " + " + textPadding + ")", borderRadius: react_theme_1.tokens.borderRadiusCircular, 
        // Use a transparent stroke (rather than no border) so the border is visible in high contrast
        borderColor: react_theme_1.tokens.colorTransparentStroke, '::after': {
            content: '""',
            position: 'absolute',
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
            borderStyle: 'solid',
            borderColor: 'inherit',
            borderWidth: react_theme_1.tokens.strokeWidthThin,
            borderRadius: 'inherit',
        } }));
    var useRootStyles = react_1.makeStyles({
        fontSmallToTiny: tslib_1.__assign({}, react_theme_1.typographyStyles.caption2Strong),
        // size
        tiny: tslib_1.__assign({ width: '6px', height: '6px', fontSize: '4px', lineHeight: '4px', minWidth: 'unset' }, react_1.shorthands.padding('unset')),
        'extra-small': tslib_1.__assign({ width: '10px', height: '10px', fontSize: '6px', lineHeight: '6px', minWidth: 'unset' }, react_1.shorthands.padding('unset')),
        small: tslib_1.__assign({ minWidth: '16px', height: '16px' }, react_1.shorthands.padding(0, "calc(" + react_theme_1.tokens.spacingHorizontalXXS + " + " + textPadding + ")")),
        medium: {
        // Set by useRootClassName
        },
        large: tslib_1.__assign({ minWidth: '24px', height: '24px' }, react_1.shorthands.padding(0, "calc(" + react_theme_1.tokens.spacingHorizontalXS + " + " + textPadding + ")")),
        'extra-large': tslib_1.__assign({ minWidth: '32px', height: '32px' }, react_1.shorthands.padding(0, "calc(" + react_theme_1.tokens.spacingHorizontalSNudge + " + " + textPadding + ")")),
        // shape
        square: tslib_1.__assign({}, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusNone)),
        rounded: tslib_1.__assign({}, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusMedium)),
        roundedSmallToTiny: tslib_1.__assign({}, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusSmall)),
        circular: {
        // Set by useRootClassName
        },
        // hide the boder when appearance is "ghost"
        borderGhost: {
            // The border is applied in an ::after pseudo-element because it should not affect layout.
            // The padding and size of the badge should be the same regardless of whether or not it has a border.
            '::after': {
                display: 'none',
            },
        },
        // appearance: filled
        filled: {
        // Set by useRootClassName
        },
        'filled-brand': {
            backgroundColor: react_theme_1.tokens.colorBrandBackground,
            color: react_theme_1.tokens.colorNeutralForegroundOnBrand,
        },
        'filled-danger': {
            backgroundColor: react_theme_1.tokens.colorPaletteRedBackground3,
            color: react_theme_1.tokens.colorNeutralForegroundOnBrand,
        },
        'filled-important': {
            backgroundColor: react_theme_1.tokens.colorNeutralForeground1,
            color: react_theme_1.tokens.colorNeutralBackground1,
        },
        'filled-informative': {
            backgroundColor: react_theme_1.tokens.colorNeutralBackground5,
            color: react_theme_1.tokens.colorNeutralForeground3,
        },
        'filled-severe': {
            backgroundColor: react_theme_1.tokens.colorPaletteDarkOrangeBackground3,
            color: react_theme_1.tokens.colorNeutralForegroundOnBrand,
        },
        'filled-subtle': {
            backgroundColor: react_theme_1.tokens.colorNeutralBackground1,
            color: react_theme_1.tokens.colorNeutralForeground1,
        },
        'filled-success': {
            backgroundColor: react_theme_1.tokens.colorPaletteGreenBackground3,
            color: react_theme_1.tokens.colorNeutralForegroundOnBrand,
        },
        'filled-warning': {
            backgroundColor: react_theme_1.tokens.colorPaletteYellowBackground3,
            color: react_theme_1.tokens.colorNeutralForeground1Static,
        },
        // appearance: ghost
        ghost: {
        // No shared colors between ghost appearances
        },
        'ghost-brand': {
            color: react_theme_1.tokens.colorBrandForeground1,
        },
        'ghost-danger': {
            color: react_theme_1.tokens.colorPaletteRedForeground3,
        },
        'ghost-important': {
            color: react_theme_1.tokens.colorNeutralForeground1,
        },
        'ghost-informative': {
            color: react_theme_1.tokens.colorNeutralForeground3,
        },
        'ghost-severe': {
            color: react_theme_1.tokens.colorPaletteDarkOrangeForeground3,
        },
        'ghost-subtle': {
            color: react_theme_1.tokens.colorNeutralForegroundStaticInverted,
        },
        'ghost-success': {
            color: react_theme_1.tokens.colorPaletteGreenForeground3,
        },
        'ghost-warning': {
            color: react_theme_1.tokens.colorPaletteYellowForeground2,
        },
        // appearance: outline
        outline: tslib_1.__assign({}, react_1.shorthands.borderColor('currentColor')),
        'outline-brand': {
            color: react_theme_1.tokens.colorBrandForeground1,
        },
        'outline-danger': tslib_1.__assign({ color: react_theme_1.tokens.colorPaletteRedForeground3 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteRedBorder2)),
        'outline-important': tslib_1.__assign({ color: react_theme_1.tokens.colorNeutralForeground3 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStrokeAccessible)),
        'outline-informative': tslib_1.__assign({ color: react_theme_1.tokens.colorNeutralForeground3 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStroke2)),
        'outline-severe': {
            color: react_theme_1.tokens.colorPaletteDarkOrangeForeground3,
        },
        'outline-subtle': {
            color: react_theme_1.tokens.colorNeutralForegroundStaticInverted,
        },
        'outline-success': tslib_1.__assign({ color: react_theme_1.tokens.colorPaletteGreenForeground3 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteGreenBorder2)),
        'outline-warning': {
            color: react_theme_1.tokens.colorPaletteYellowForeground2,
        },
        // appearance: tint
        tint: {
        // No shared colors between tint appearances
        },
        'tint-brand': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorBrandBackground2, color: react_theme_1.tokens.colorBrandForeground2 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorBrandStroke2)),
        'tint-danger': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorPaletteRedBackground1, color: react_theme_1.tokens.colorPaletteRedForeground1 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteRedBorder1)),
        'tint-important': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorNeutralForeground3, color: react_theme_1.tokens.colorNeutralBackground1 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorTransparentStroke)),
        'tint-informative': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorNeutralBackground4, color: react_theme_1.tokens.colorNeutralForeground3 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStroke2)),
        'tint-severe': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorPaletteDarkOrangeBackground1, color: react_theme_1.tokens.colorPaletteDarkOrangeForeground1 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteDarkOrangeBorder1)),
        'tint-subtle': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorNeutralBackground1, color: react_theme_1.tokens.colorNeutralForeground3 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStroke2)),
        'tint-success': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorPaletteGreenBackground1, color: react_theme_1.tokens.colorPaletteGreenForeground1 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteGreenBorder1)),
        'tint-warning': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorPaletteYellowBackground1, color: react_theme_1.tokens.colorPaletteYellowForeground1 }, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteYellowBorder1)),
    });
    var useIconRootClassName = react_1.makeResetStyles({
        display: 'flex',
        lineHeight: '1',
        margin: "0 calc(-1 * " + textPadding + ")",
        fontSize: '12px',
    });
    var useIconStyles = react_1.makeStyles({
        beforeText: {
            marginRight: "calc(" + react_theme_1.tokens.spacingHorizontalXXS + " + " + textPadding + ")",
        },
        afterText: {
            marginLeft: "calc(" + react_theme_1.tokens.spacingHorizontalXXS + " + " + textPadding + ")",
        },
        beforeTextXL: {
            marginRight: "calc(" + react_theme_1.tokens.spacingHorizontalXS + " + " + textPadding + ")",
        },
        afterTextXL: {
            marginLeft: "calc(" + react_theme_1.tokens.spacingHorizontalXS + " + " + textPadding + ")",
        },
        // size
        tiny: {
            fontSize: '6px',
        },
        'extra-small': {
            fontSize: '10px',
        },
        small: {
            fontSize: '12px',
        },
        medium: {
        // Set by useIconRootClassName
        },
        large: {
            fontSize: '16px',
        },
        'extra-large': {
            fontSize: '20px',
        },
    });
    /**
     * Applies style classnames to slots
     */
    var useBadgeStyles_unstable = function (state) {
        var rootClassName = useRootClassName();
        var rootStyles = useRootStyles();
        var smallToTiny = state.size === 'small' || state.size === 'extra-small' || state.size === 'tiny';
        state.root.className = react_1.mergeClasses(exports.badgeClassNames.root, rootClassName, smallToTiny && rootStyles.fontSmallToTiny, rootStyles[state.size], rootStyles[state.shape], state.shape === 'rounded' && smallToTiny && rootStyles.roundedSmallToTiny, state.appearance === 'ghost' && rootStyles.borderGhost, rootStyles[state.appearance], rootStyles[state.appearance + "-" + state.color], state.root.className);
        var iconRootClassName = useIconRootClassName();
        var iconStyles = useIconStyles();
        if (state.icon) {
            var iconPositionClass = void 0;
            if (state.root.children) {
                if (state.size === 'extra-large') {
                    iconPositionClass = state.iconPosition === 'after' ? iconStyles.afterTextXL : iconStyles.beforeTextXL;
                }
                else {
                    iconPositionClass = state.iconPosition === 'after' ? iconStyles.afterText : iconStyles.beforeText;
                }
            }
            state.icon.className = react_1.mergeClasses(exports.badgeClassNames.icon, iconRootClassName, iconPositionClass, iconStyles[state.size], state.icon.className);
        }
        return state;
    };
    exports.useBadgeStyles_unstable = useBadgeStyles_unstable;
});
