// @fluentui/react-tabs/lib-commonjs/components/TabList/TabListContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useTabListContext_unstable = exports.TabListProvider = exports.TabListContext = void 0;
const react_context_selector_1 = /*#__PURE__*/require("@fluentui/react-context-selector");
const tabListContextDefaultValue = {
  appearance: 'transparent',
  reserveSelectedTabSpace: true,
  disabled: false,
  selectedValue: undefined,
  onRegister: () => {
    /* noop */
  },
  onUnregister: () => {
    /* noop */
  },
  onSelect: () => {
    /* noop */
  },
  getRegisteredTabs: () => {
    return {
      registeredTabs: {}
    };
  },
  size: 'medium',
  vertical: false
};
exports.TabListContext = /*#__PURE__*/react_context_selector_1.createContext(undefined);
exports.TabListProvider = exports.TabListContext.Provider;
const useTabListContext_unstable = selector => react_context_selector_1.useContextSelector(exports.TabListContext, (ctx = tabListContextDefaultValue) => selector(ctx));
exports.useTabListContext_unstable = useTabListContext_unstable;
