// @fluentui/react-text/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.title3ClassNames = exports.Title3 = exports.title2ClassNames = exports.Title2 = exports.title1ClassNames = exports.Title1 = exports.subtitle2StrongerClassNames = exports.Subtitle2Stronger = exports.subtitle2ClassNames = exports.Subtitle2 = exports.subtitle1ClassNames = exports.Subtitle1 = exports.largeTitleClassNames = exports.LargeTitle = exports.displayClassNames = exports.Display = exports.caption2StrongClassNames = exports.Caption2Strong = exports.caption2ClassNames = exports.Caption2 = exports.caption1StrongerClassNames = exports.Caption1Stronger = exports.caption1StrongClassNames = exports.Caption1Strong = exports.caption1ClassNames = exports.Caption1 = exports.body2ClassNames = exports.Body2 = exports.body1StrongerClassNames = exports.Body1Stronger = exports.body1StrongClassNames = exports.Body1Strong = exports.body1ClassNames = exports.Body1 = exports.useText_unstable = exports.useTextStyles_unstable = exports.textClassNames = exports.renderText_unstable = exports.Text = void 0;
var Text_1 = /*#__PURE__*/require("./Text");
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function () {
    return Text_1.Text;
  }
});
Object.defineProperty(exports, "renderText_unstable", {
  enumerable: true,
  get: function () {
    return Text_1.renderText_unstable;
  }
});
Object.defineProperty(exports, "textClassNames", {
  enumerable: true,
  get: function () {
    return Text_1.textClassNames;
  }
});
Object.defineProperty(exports, "useTextStyles_unstable", {
  enumerable: true,
  get: function () {
    return Text_1.useTextStyles_unstable;
  }
});
Object.defineProperty(exports, "useText_unstable", {
  enumerable: true,
  get: function () {
    return Text_1.useText_unstable;
  }
});
var Body1_1 = /*#__PURE__*/require("./Body1");
Object.defineProperty(exports, "Body1", {
  enumerable: true,
  get: function () {
    return Body1_1.Body1;
  }
});
Object.defineProperty(exports, "body1ClassNames", {
  enumerable: true,
  get: function () {
    return Body1_1.body1ClassNames;
  }
});
var Body1Strong_1 = /*#__PURE__*/require("./Body1Strong");
Object.defineProperty(exports, "Body1Strong", {
  enumerable: true,
  get: function () {
    return Body1Strong_1.Body1Strong;
  }
});
Object.defineProperty(exports, "body1StrongClassNames", {
  enumerable: true,
  get: function () {
    return Body1Strong_1.body1StrongClassNames;
  }
});
var Body1Stronger_1 = /*#__PURE__*/require("./Body1Stronger");
Object.defineProperty(exports, "Body1Stronger", {
  enumerable: true,
  get: function () {
    return Body1Stronger_1.Body1Stronger;
  }
});
Object.defineProperty(exports, "body1StrongerClassNames", {
  enumerable: true,
  get: function () {
    return Body1Stronger_1.body1StrongerClassNames;
  }
});
var Body2_1 = /*#__PURE__*/require("./Body2");
Object.defineProperty(exports, "Body2", {
  enumerable: true,
  get: function () {
    return Body2_1.Body2;
  }
});
Object.defineProperty(exports, "body2ClassNames", {
  enumerable: true,
  get: function () {
    return Body2_1.body2ClassNames;
  }
});
var Caption1_1 = /*#__PURE__*/require("./Caption1");
Object.defineProperty(exports, "Caption1", {
  enumerable: true,
  get: function () {
    return Caption1_1.Caption1;
  }
});
Object.defineProperty(exports, "caption1ClassNames", {
  enumerable: true,
  get: function () {
    return Caption1_1.caption1ClassNames;
  }
});
var Caption1Strong_1 = /*#__PURE__*/require("./Caption1Strong");
Object.defineProperty(exports, "Caption1Strong", {
  enumerable: true,
  get: function () {
    return Caption1Strong_1.Caption1Strong;
  }
});
Object.defineProperty(exports, "caption1StrongClassNames", {
  enumerable: true,
  get: function () {
    return Caption1Strong_1.caption1StrongClassNames;
  }
});
var Caption1Stronger_1 = /*#__PURE__*/require("./Caption1Stronger");
Object.defineProperty(exports, "Caption1Stronger", {
  enumerable: true,
  get: function () {
    return Caption1Stronger_1.Caption1Stronger;
  }
});
Object.defineProperty(exports, "caption1StrongerClassNames", {
  enumerable: true,
  get: function () {
    return Caption1Stronger_1.caption1StrongerClassNames;
  }
});
var Caption2_1 = /*#__PURE__*/require("./Caption2");
Object.defineProperty(exports, "Caption2", {
  enumerable: true,
  get: function () {
    return Caption2_1.Caption2;
  }
});
Object.defineProperty(exports, "caption2ClassNames", {
  enumerable: true,
  get: function () {
    return Caption2_1.caption2ClassNames;
  }
});
var Caption2Strong_1 = /*#__PURE__*/require("./Caption2Strong");
Object.defineProperty(exports, "Caption2Strong", {
  enumerable: true,
  get: function () {
    return Caption2Strong_1.Caption2Strong;
  }
});
Object.defineProperty(exports, "caption2StrongClassNames", {
  enumerable: true,
  get: function () {
    return Caption2Strong_1.caption2StrongClassNames;
  }
});
var Display_1 = /*#__PURE__*/require("./Display");
Object.defineProperty(exports, "Display", {
  enumerable: true,
  get: function () {
    return Display_1.Display;
  }
});
Object.defineProperty(exports, "displayClassNames", {
  enumerable: true,
  get: function () {
    return Display_1.displayClassNames;
  }
});
var LargeTitle_1 = /*#__PURE__*/require("./LargeTitle");
Object.defineProperty(exports, "LargeTitle", {
  enumerable: true,
  get: function () {
    return LargeTitle_1.LargeTitle;
  }
});
Object.defineProperty(exports, "largeTitleClassNames", {
  enumerable: true,
  get: function () {
    return LargeTitle_1.largeTitleClassNames;
  }
});
var Subtitle1_1 = /*#__PURE__*/require("./Subtitle1");
Object.defineProperty(exports, "Subtitle1", {
  enumerable: true,
  get: function () {
    return Subtitle1_1.Subtitle1;
  }
});
Object.defineProperty(exports, "subtitle1ClassNames", {
  enumerable: true,
  get: function () {
    return Subtitle1_1.subtitle1ClassNames;
  }
});
var Subtitle2_1 = /*#__PURE__*/require("./Subtitle2");
Object.defineProperty(exports, "Subtitle2", {
  enumerable: true,
  get: function () {
    return Subtitle2_1.Subtitle2;
  }
});
Object.defineProperty(exports, "subtitle2ClassNames", {
  enumerable: true,
  get: function () {
    return Subtitle2_1.subtitle2ClassNames;
  }
});
var Subtitle2Stronger_1 = /*#__PURE__*/require("./Subtitle2Stronger");
Object.defineProperty(exports, "Subtitle2Stronger", {
  enumerable: true,
  get: function () {
    return Subtitle2Stronger_1.Subtitle2Stronger;
  }
});
Object.defineProperty(exports, "subtitle2StrongerClassNames", {
  enumerable: true,
  get: function () {
    return Subtitle2Stronger_1.subtitle2StrongerClassNames;
  }
});
var Title1_1 = /*#__PURE__*/require("./Title1");
Object.defineProperty(exports, "Title1", {
  enumerable: true,
  get: function () {
    return Title1_1.Title1;
  }
});
Object.defineProperty(exports, "title1ClassNames", {
  enumerable: true,
  get: function () {
    return Title1_1.title1ClassNames;
  }
});
var Title2_1 = /*#__PURE__*/require("./Title2");
Object.defineProperty(exports, "Title2", {
  enumerable: true,
  get: function () {
    return Title2_1.Title2;
  }
});
Object.defineProperty(exports, "title2ClassNames", {
  enumerable: true,
  get: function () {
    return Title2_1.title2ClassNames;
  }
});
var Title3_1 = /*#__PURE__*/require("./Title3");
Object.defineProperty(exports, "Title3", {
  enumerable: true,
  get: function () {
    return Title3_1.Title3;
  }
});
Object.defineProperty(exports, "title3ClassNames", {
  enumerable: true,
  get: function () {
    return Title3_1.title3ClassNames;
  }
});
