// @fluentui/react-combobox/lib-amd/components/Combobox/useComboboxStyles.js
define(["require", "exports", "tslib", "@fluentui/react-theme", "@griffel/react", "../../utils/internalTokens"], function (require, exports, tslib_1, react_theme_1, react_1, internalTokens_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useComboboxStyles_unstable = exports.comboboxClassNames = void 0;
    exports.comboboxClassNames = {
        root: 'fui-Combobox',
        input: 'fui-Combobox__input',
        expandIcon: 'fui-Combobox__expandIcon',
        listbox: 'fui-Combobox__listbox',
    };
    // Matches internal heights for Select and Input, but there are no theme variables for these
    // field heights are 2px less than other controls, since the border is on the parent element.
    var fieldHeights = {
        small: '22px',
        medium: '30px',
        large: '38px',
    };
    /**
     * Styles for Combobox
     */
    var useStyles = react_1.makeStyles({
        root: tslib_1.__assign(tslib_1.__assign({ alignItems: 'center' }, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusMedium)), { boxSizing: 'border-box', columnGap: react_theme_1.tokens.spacingHorizontalXXS, display: 'inline-grid', gridTemplateColumns: '1fr auto', justifyContent: 'space-between', minWidth: '250px', position: 'relative', 
            // windows high contrast mode focus indicator
            ':focus-within': {
                outlineWidth: '2px',
                outlineStyle: 'solid',
                outlineColor: 'transparent',
            }, 
            // bottom focus border, shared with Input, Select, and SpinButton
            '::after': tslib_1.__assign(tslib_1.__assign({ boxSizing: 'border-box', content: '""', position: 'absolute', left: '-1px', bottom: '-1px', right: '-1px', height: "max(2px, " + react_theme_1.tokens.borderRadiusMedium + ")", borderBottomLeftRadius: react_theme_1.tokens.borderRadiusMedium, borderBottomRightRadius: react_theme_1.tokens.borderRadiusMedium }, react_1.shorthands.borderBottom(react_theme_1.tokens.strokeWidthThick, 'solid', react_theme_1.tokens.colorCompoundBrandStroke)), { clipPath: 'inset(calc(100% - 2px) 0 0 0)', transform: 'scaleX(0)', transitionProperty: 'transform', transitionDuration: react_theme_1.tokens.durationUltraFast, transitionDelay: react_theme_1.tokens.curveAccelerateMid, '@media screen and (prefers-reduced-motion: reduce)': {
                    transitionDuration: '0.01ms',
                    transitionDelay: '0.01ms',
                } }), ':focus-within::after': {
                transform: 'scaleX(1)',
                transitionProperty: 'transform',
                transitionDuration: react_theme_1.tokens.durationNormal,
                transitionDelay: react_theme_1.tokens.curveDecelerateMid,
                '@media screen and (prefers-reduced-motion: reduce)': {
                    transitionDuration: '0.01ms',
                    transitionDelay: '0.01ms',
                },
            }, ':focus-within:active::after': {
                borderBottomColor: react_theme_1.tokens.colorCompoundBrandStrokePressed,
            } }),
        listbox: tslib_1.__assign(tslib_1.__assign({ boxShadow: "" + react_theme_1.tokens.shadow16 }, react_1.shorthands.borderRadius(react_theme_1.tokens.borderRadiusMedium)), { maxHeight: '80vh' }),
        listboxCollapsed: {
            display: 'none',
        },
        // size variants
        small: {
            paddingRight: react_theme_1.tokens.spacingHorizontalSNudge,
        },
        medium: {
            paddingRight: react_theme_1.tokens.spacingHorizontalMNudge,
        },
        large: {
            columnGap: react_theme_1.tokens.spacingHorizontalSNudge,
            paddingRight: react_theme_1.tokens.spacingHorizontalM,
        },
        // appearance variants
        outline: tslib_1.__assign(tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorNeutralBackground1 }, react_1.shorthands.border(react_theme_1.tokens.strokeWidthThin, 'solid', react_theme_1.tokens.colorNeutralStroke1)), { borderBottomColor: react_theme_1.tokens.colorNeutralStrokeAccessible }),
        outlineInteractive: {
            '&:hover': tslib_1.__assign(tslib_1.__assign({}, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStroke1Hover)), { borderBottomColor: react_theme_1.tokens.colorNeutralStrokeAccessible }),
            '&:active': tslib_1.__assign(tslib_1.__assign({}, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStroke1Pressed)), { borderBottomColor: react_theme_1.tokens.colorNeutralStrokeAccessible }),
        },
        underline: tslib_1.__assign(tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorTransparentBackground }, react_1.shorthands.borderBottom(react_theme_1.tokens.strokeWidthThin, 'solid', react_theme_1.tokens.colorNeutralStrokeAccessible)), react_1.shorthands.borderRadius(0)),
        'filled-lighter': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorNeutralBackground1 }, react_1.shorthands.border(react_theme_1.tokens.strokeWidthThin, 'solid', react_theme_1.tokens.colorTransparentStroke)),
        'filled-darker': tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorNeutralBackground3 }, react_1.shorthands.border(react_theme_1.tokens.strokeWidthThin, 'solid', react_theme_1.tokens.colorTransparentStroke)),
        invalid: {
            ':not(:focus-within),:hover:not(:focus-within)': tslib_1.__assign({}, react_1.shorthands.borderColor(react_theme_1.tokens.colorPaletteRedBorder2)),
        },
        invalidUnderline: {
            ':not(:focus-within),:hover:not(:focus-within)': {
                borderBottomColor: react_theme_1.tokens.colorPaletteRedBorder2,
            },
        },
        disabled: tslib_1.__assign(tslib_1.__assign({ cursor: 'not-allowed', backgroundColor: react_theme_1.tokens.colorTransparentBackground }, react_1.shorthands.borderColor(react_theme_1.tokens.colorNeutralStrokeDisabled)), { '@media (forced-colors: active)': tslib_1.__assign({}, react_1.shorthands.borderColor('GrayText')) }),
    });
    var useInputStyles = react_1.makeStyles({
        input: tslib_1.__assign(tslib_1.__assign({ backgroundColor: react_theme_1.tokens.colorTransparentBackground }, react_1.shorthands.border('0')), { color: react_theme_1.tokens.colorNeutralForeground1, fontFamily: react_theme_1.tokens.fontFamilyBase, '&:focus': {
                outlineStyle: 'none',
            }, '&::placeholder': {
                color: react_theme_1.tokens.colorNeutralForeground4,
                opacity: 1,
            } }),
        // size variants
        small: tslib_1.__assign(tslib_1.__assign({ height: fieldHeights.small }, react_theme_1.typographyStyles.caption1), react_1.shorthands.padding(0, 0, 0, "calc(" + react_theme_1.tokens.spacingHorizontalSNudge + " + " + react_theme_1.tokens.spacingHorizontalXXS + ")")),
        medium: tslib_1.__assign(tslib_1.__assign({ height: fieldHeights.medium }, react_theme_1.typographyStyles.body1), react_1.shorthands.padding(0, 0, 0, "calc(" + react_theme_1.tokens.spacingHorizontalMNudge + " + " + react_theme_1.tokens.spacingHorizontalXXS + ")")),
        large: tslib_1.__assign(tslib_1.__assign({ height: fieldHeights.large }, react_theme_1.typographyStyles.body2), react_1.shorthands.padding(0, 0, 0, "calc(" + react_theme_1.tokens.spacingHorizontalM + " + " + react_theme_1.tokens.spacingHorizontalSNudge + ")")),
        disabled: {
            color: react_theme_1.tokens.colorNeutralForegroundDisabled,
            backgroundColor: react_theme_1.tokens.colorTransparentBackground,
            cursor: 'not-allowed',
            '::placeholder': {
                color: react_theme_1.tokens.colorNeutralForegroundDisabled,
            },
        },
    });
    var useIconStyles = react_1.makeStyles({
        icon: {
            boxSizing: 'border-box',
            color: react_theme_1.tokens.colorNeutralStrokeAccessible,
            cursor: 'pointer',
            display: 'block',
            fontSize: react_theme_1.tokens.fontSizeBase500,
            // the SVG must have display: block for accurate positioning
            // otherwise an extra inline space is inserted after the svg element
            '& svg': {
                display: 'block',
            },
        },
        // icon size variants
        small: {
            fontSize: internalTokens_1.iconSizes.small,
            marginLeft: react_theme_1.tokens.spacingHorizontalXXS,
        },
        medium: {
            fontSize: internalTokens_1.iconSizes.medium,
            marginLeft: react_theme_1.tokens.spacingHorizontalXXS,
        },
        large: {
            fontSize: internalTokens_1.iconSizes.large,
            marginLeft: react_theme_1.tokens.spacingHorizontalSNudge,
        },
        disabled: {
            color: react_theme_1.tokens.colorNeutralForegroundDisabled,
            cursor: 'not-allowed',
        },
    });
    /**
     * Apply styling to the Combobox slots based on the state
     */
    var useComboboxStyles_unstable = function (state) {
        var appearance = state.appearance, open = state.open, size = state.size;
        var invalid = "" + state.input['aria-invalid'] === 'true';
        var disabled = state.input.disabled;
        var styles = useStyles();
        var iconStyles = useIconStyles();
        var inputStyles = useInputStyles();
        state.root.className = react_1.mergeClasses(exports.comboboxClassNames.root, styles.root, styles[appearance], styles[size], !disabled && appearance === 'outline' && styles.outlineInteractive, invalid && appearance !== 'underline' && styles.invalid, invalid && appearance === 'underline' && styles.invalidUnderline, disabled && styles.disabled, state.root.className);
        state.input.className = react_1.mergeClasses(exports.comboboxClassNames.input, inputStyles.input, inputStyles[size], disabled && inputStyles.disabled, state.input.className);
        if (state.listbox) {
            state.listbox.className = react_1.mergeClasses(exports.comboboxClassNames.listbox, styles.listbox, !open && styles.listboxCollapsed, state.listbox.className);
        }
        if (state.expandIcon) {
            state.expandIcon.className = react_1.mergeClasses(exports.comboboxClassNames.expandIcon, iconStyles.icon, iconStyles[size], disabled && iconStyles.disabled, state.expandIcon.className);
        }
        return state;
    };
    exports.useComboboxStyles_unstable = useComboboxStyles_unstable;
});
