// @fluentui/react-components/lib-commonjs/unstable/index.js
"use strict";

// Stub for unstable exports
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.switchFieldClassNames = exports.SwitchField = exports.spinButtonFieldClassNames = exports.SpinButtonField = exports.sliderFieldClassNames = exports.SliderField = exports.selectFieldClassNames = exports.SelectField = exports.radioGroupFieldClassNames = exports.RadioGroupField = exports.progressFieldClassNames = exports.ProgressField = exports.inputFieldClassNames = exports.InputField = exports.comboboxFieldClassNames = exports.ComboboxField = exports.checkboxFieldClassNames = exports.CheckboxField = exports.renderInfoButton_unstable = exports.useInfoButtonStyles_unstable = exports.useInfoButton_unstable = exports.infoButtonClassNames = exports.InfoButton = exports.useCardStyles_unstable = exports.useCardPreviewStyles_unstable = exports.useCardPreview_unstable = exports.useCardHeaderStyles_unstable = exports.useCardHeader_unstable = exports.useCardFooterStyles_unstable = exports.useCardFooter_unstable = exports.useCard_unstable = exports.renderCardPreview_unstable = exports.renderCardHeader_unstable = exports.renderCardFooter_unstable = exports.renderCard_unstable = exports.cardPreviewClassNames = exports.cardHeaderCSSVars = exports.cardHeaderClassNames = exports.cardFooterClassNames = exports.cardCSSVars = exports.cardClassNames = exports.CardPreview = exports.CardHeader = exports.CardFooter = exports.Card = exports.useAlert_unstable = exports.useAlertStyles_unstable = exports.renderAlert_unstable = exports.alertClassNames = exports.Alert = void 0;
exports.useIntersectionObserver = exports.useVirtualizerStyles_unstable = exports.renderVirtualizer_unstable = exports.useVirtualizer_unstable = exports.virtualizerClassNames = exports.Virtualizer = exports.useField_unstable = exports.useFieldStyles_unstable = exports.renderField_unstable = exports.fieldClassNames = exports.Field = exports.textareaFieldClassNames = exports.TextareaField = void 0;
var react_alert_1 = /*#__PURE__*/require("@fluentui/react-alert");
Object.defineProperty(exports, "Alert", {
  enumerable: true,
  get: function () {
    return react_alert_1.Alert;
  }
});
Object.defineProperty(exports, "alertClassNames", {
  enumerable: true,
  get: function () {
    return react_alert_1.alertClassNames;
  }
});
Object.defineProperty(exports, "renderAlert_unstable", {
  enumerable: true,
  get: function () {
    return react_alert_1.renderAlert_unstable;
  }
});
Object.defineProperty(exports, "useAlertStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_alert_1.useAlertStyles_unstable;
  }
});
Object.defineProperty(exports, "useAlert_unstable", {
  enumerable: true,
  get: function () {
    return react_alert_1.useAlert_unstable;
  }
});
var react_card_1 = /*#__PURE__*/require("@fluentui/react-card");
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function () {
    return react_card_1.Card;
  }
});
Object.defineProperty(exports, "CardFooter", {
  enumerable: true,
  get: function () {
    return react_card_1.CardFooter;
  }
});
Object.defineProperty(exports, "CardHeader", {
  enumerable: true,
  get: function () {
    return react_card_1.CardHeader;
  }
});
Object.defineProperty(exports, "CardPreview", {
  enumerable: true,
  get: function () {
    return react_card_1.CardPreview;
  }
});
Object.defineProperty(exports, "cardClassNames", {
  enumerable: true,
  get: function () {
    return react_card_1.cardClassNames;
  }
});
Object.defineProperty(exports, "cardCSSVars", {
  enumerable: true,
  get: function () {
    return react_card_1.cardCSSVars;
  }
});
Object.defineProperty(exports, "cardFooterClassNames", {
  enumerable: true,
  get: function () {
    return react_card_1.cardFooterClassNames;
  }
});
Object.defineProperty(exports, "cardHeaderClassNames", {
  enumerable: true,
  get: function () {
    return react_card_1.cardHeaderClassNames;
  }
});
Object.defineProperty(exports, "cardHeaderCSSVars", {
  enumerable: true,
  get: function () {
    return react_card_1.cardHeaderCSSVars;
  }
});
Object.defineProperty(exports, "cardPreviewClassNames", {
  enumerable: true,
  get: function () {
    return react_card_1.cardPreviewClassNames;
  }
});
Object.defineProperty(exports, "renderCard_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.renderCard_unstable;
  }
});
Object.defineProperty(exports, "renderCardFooter_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.renderCardFooter_unstable;
  }
});
Object.defineProperty(exports, "renderCardHeader_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.renderCardHeader_unstable;
  }
});
Object.defineProperty(exports, "renderCardPreview_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.renderCardPreview_unstable;
  }
});
Object.defineProperty(exports, "useCard_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCard_unstable;
  }
});
Object.defineProperty(exports, "useCardFooter_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardFooter_unstable;
  }
});
Object.defineProperty(exports, "useCardFooterStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardFooterStyles_unstable;
  }
});
Object.defineProperty(exports, "useCardHeader_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardHeader_unstable;
  }
});
Object.defineProperty(exports, "useCardHeaderStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardHeaderStyles_unstable;
  }
});
Object.defineProperty(exports, "useCardPreview_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardPreview_unstable;
  }
});
Object.defineProperty(exports, "useCardPreviewStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardPreviewStyles_unstable;
  }
});
Object.defineProperty(exports, "useCardStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_card_1.useCardStyles_unstable;
  }
});
var react_infobutton_1 = /*#__PURE__*/require("@fluentui/react-infobutton");
Object.defineProperty(exports, "InfoButton", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.InfoButton;
  }
});
Object.defineProperty(exports, "infoButtonClassNames", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.infoButtonClassNames;
  }
});
Object.defineProperty(exports, "useInfoButton_unstable", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.useInfoButton_unstable;
  }
});
Object.defineProperty(exports, "useInfoButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.useInfoButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "renderInfoButton_unstable", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.renderInfoButton_unstable;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_checkbox_1 = /*#__PURE__*/require("@fluentui/react-checkbox");
Object.defineProperty(exports, "CheckboxField", {
  enumerable: true,
  get: function () {
    return react_checkbox_1.CheckboxField_unstable;
  }
});
Object.defineProperty(exports, "checkboxFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_checkbox_1.checkboxFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_combobox_1 = /*#__PURE__*/require("@fluentui/react-combobox");
Object.defineProperty(exports, "ComboboxField", {
  enumerable: true,
  get: function () {
    return react_combobox_1.ComboboxField_unstable;
  }
});
Object.defineProperty(exports, "comboboxFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_combobox_1.comboboxFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_input_1 = /*#__PURE__*/require("@fluentui/react-input");
Object.defineProperty(exports, "InputField", {
  enumerable: true,
  get: function () {
    return react_input_1.InputField_unstable;
  }
});
Object.defineProperty(exports, "inputFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_input_1.inputFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_progress_1 = /*#__PURE__*/require("@fluentui/react-progress");
Object.defineProperty(exports, "ProgressField", {
  enumerable: true,
  get: function () {
    return react_progress_1.ProgressField_unstable;
  }
});
Object.defineProperty(exports, "progressFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_progress_1.progressFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_radio_1 = /*#__PURE__*/require("@fluentui/react-radio");
Object.defineProperty(exports, "RadioGroupField", {
  enumerable: true,
  get: function () {
    return react_radio_1.RadioGroupField_unstable;
  }
});
Object.defineProperty(exports, "radioGroupFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_radio_1.radioGroupFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_select_1 = /*#__PURE__*/require("@fluentui/react-select");
Object.defineProperty(exports, "SelectField", {
  enumerable: true,
  get: function () {
    return react_select_1.SelectField_unstable;
  }
});
Object.defineProperty(exports, "selectFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_select_1.selectFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_slider_1 = /*#__PURE__*/require("@fluentui/react-slider");
Object.defineProperty(exports, "SliderField", {
  enumerable: true,
  get: function () {
    return react_slider_1.SliderField_unstable;
  }
});
Object.defineProperty(exports, "sliderFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_slider_1.sliderFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_spinbutton_1 = /*#__PURE__*/require("@fluentui/react-spinbutton");
Object.defineProperty(exports, "SpinButtonField", {
  enumerable: true,
  get: function () {
    return react_spinbutton_1.SpinButtonField_unstable;
  }
});
Object.defineProperty(exports, "spinButtonFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_spinbutton_1.spinButtonFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_switch_1 = /*#__PURE__*/require("@fluentui/react-switch");
Object.defineProperty(exports, "SwitchField", {
  enumerable: true,
  get: function () {
    return react_switch_1.SwitchField_unstable;
  }
});
Object.defineProperty(exports, "switchFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_switch_1.switchFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_textarea_1 = /*#__PURE__*/require("@fluentui/react-textarea");
Object.defineProperty(exports, "TextareaField", {
  enumerable: true,
  get: function () {
    return react_textarea_1.TextareaField_unstable;
  }
});
Object.defineProperty(exports, "textareaFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_textarea_1.textareaFieldClassNames;
  }
});
var react_field_1 = /*#__PURE__*/require("@fluentui/react-field");
Object.defineProperty(exports, "Field", {
  enumerable: true,
  get: function () {
    return react_field_1.Field;
  }
});
Object.defineProperty(exports, "fieldClassNames", {
  enumerable: true,
  get: function () {
    return react_field_1.fieldClassNames;
  }
});
Object.defineProperty(exports, "renderField_unstable", {
  enumerable: true,
  get: function () {
    return react_field_1.renderField_unstable;
  }
});
Object.defineProperty(exports, "useFieldStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_field_1.useFieldStyles_unstable;
  }
});
Object.defineProperty(exports, "useField_unstable", {
  enumerable: true,
  get: function () {
    return react_field_1.useField_unstable;
  }
});
var react_virtualizer_1 = /*#__PURE__*/require("@fluentui/react-virtualizer");
Object.defineProperty(exports, "Virtualizer", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.Virtualizer;
  }
});
Object.defineProperty(exports, "virtualizerClassNames", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.virtualizerClassNames;
  }
});
Object.defineProperty(exports, "useVirtualizer_unstable", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.useVirtualizer_unstable;
  }
});
Object.defineProperty(exports, "renderVirtualizer_unstable", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.renderVirtualizer_unstable;
  }
});
Object.defineProperty(exports, "useVirtualizerStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.useVirtualizerStyles_unstable;
  }
});
Object.defineProperty(exports, "useIntersectionObserver", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.useIntersectionObserver;
  }
});
