// @fluentui/react-positioning/lib-commonjs/middleware/maxSize.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.maxSize = void 0;
const dom_1 = /*#__PURE__*/require("@floating-ui/dom");
const getBoundary_1 = /*#__PURE__*/require("../utils/getBoundary");
function maxSize(autoSize, options) {
  const {
    container,
    overflowBoundary
  } = options;
  return dom_1.size({
    ...(overflowBoundary && {
      altBoundary: true,
      boundary: getBoundary_1.getBoundary(container, overflowBoundary)
    }),
    apply({
      availableHeight,
      availableWidth,
      elements,
      rects
    }) {
      const applyMaxWidth = autoSize === 'always' || autoSize === 'width-always' || rects.floating.width > availableWidth && (autoSize === true || autoSize === 'width');
      const applyMaxHeight = autoSize === 'always' || autoSize === 'height-always' || rects.floating.height > availableHeight && (autoSize === true || autoSize === 'height');
      if (applyMaxHeight) {
        Object.assign(elements.floating.style, {
          maxHeight: `${availableHeight}px`,
          boxSizing: 'border-box',
          overflowY: 'auto'
        });
      }
      if (applyMaxWidth) {
        Object.assign(elements.floating.style, {
          maxWidth: `${availableWidth}px`,
          boxSizing: 'border-box',
          overflowX: 'auto'
        });
      }
    }
  });
}
exports.maxSize = maxSize;
