// @fluentui/react-positioning/lib-commonjs/middleware/shift.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shift = void 0;
const dom_1 = /*#__PURE__*/require("@floating-ui/dom");
const index_1 = /*#__PURE__*/require("../utils/index");
/**
 * Wraps the floating UI shift middleware for easier usage of our options
 */
function shift(options) {
  const {
    hasScrollableElement,
    disableTether,
    overflowBoundary,
    container,
    overflowBoundaryPadding,
    isRtl
  } = options;
  return dom_1.shift({
    ...(hasScrollableElement && {
      boundary: 'clippingAncestors'
    }),
    ...(disableTether && {
      crossAxis: disableTether === 'all',
      limiter: dom_1.limitShift({
        crossAxis: disableTether !== 'all',
        mainAxis: false
      })
    }),
    ...(overflowBoundaryPadding && {
      padding: index_1.toFloatingUIPadding(overflowBoundaryPadding, isRtl)
    }),
    ...(overflowBoundary && {
      altBoundary: true,
      boundary: index_1.getBoundary(container, overflowBoundary)
    })
  });
}
exports.shift = shift;
