// @fluentui/react-table/lib-amd/components/TableResizeHandle/useTableResizeHandleStyles.js
define(["require", "exports", "tslib", "@griffel/react", "@fluentui/react-theme"], function (require, exports, tslib_1, react_1, react_theme_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableResizeHandleStyles_unstable = exports.tableResizeHandleClassNames = void 0;
    exports.tableResizeHandleClassNames = {
        root: 'fui-TableResizeHandle',
        // TODO: add class names for all slots on TableResizeHandleSlots.
        // Should be of the form `<slotName>: 'fui-TableResizeHandle__<slotName>`
    };
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: tslib_1.__assign(tslib_1.__assign({ position: 'absolute', right: 0, top: 0, bottom: 0, width: '16px' }, react_1.shorthands.margin(0, '-8px')), { cursor: 'col-resize', opacity: 0, transitionProperty: 'opacity', transitionDuration: '.2s', zIndex: 1, ':hover': {
                opacity: 1,
            }, ':after': {
                content: '" "',
                display: 'block',
                width: '1px',
                position: 'absolute',
                left: '50%',
                top: 0,
                bottom: 0,
                backgroundColor: react_theme_1.tokens.colorNeutralStroke1,
            } }),
    });
    /**
     * Apply styling to the TableResizeHandle slots based on the state
     */
    var useTableResizeHandleStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.tableResizeHandleClassNames.root, styles.root, state.root.className);
        return state;
    };
    exports.useTableResizeHandleStyles_unstable = useTableResizeHandleStyles_unstable;
});
