// @fluentui/react-table/lib-amd/components/TableHeader/useTableHeaderStyles.js
define(["require", "exports", "@griffel/react"], function (require, exports, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableHeaderStyles_unstable = exports.tableHeaderClassNames = exports.tableHeaderClassName = void 0;
    exports.tableHeaderClassName = 'fui-TableHeader';
    exports.tableHeaderClassNames = {
        root: 'fui-TableHeader',
    };
    var useFlexLayoutStyles = react_1.makeStyles({
        root: {
            display: 'block',
        },
    });
    var useTableLayoutStyles = react_1.makeStyles({
        root: {
            display: 'table-row-group',
        },
    });
    /**
     * Apply styling to the TableHeader slots based on the state
     */
    var useTableHeaderStyles_unstable = function (state) {
        var layoutStyles = {
            table: useTableLayoutStyles(),
            flex: useFlexLayoutStyles(),
        };
        state.root.className = react_1.mergeClasses(exports.tableHeaderClassName, state.noNativeElements ? layoutStyles.flex.root : layoutStyles.table.root, state.root.className);
        return state;
    };
    exports.useTableHeaderStyles_unstable = useTableHeaderStyles_unstable;
});
