// @fluentui/react-table/lib-amd/components/TableBody/useTableBodyStyles.js
define(["require", "exports", "@griffel/react"], function (require, exports, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableBodyStyles_unstable = exports.tableBodyClassNames = exports.tableBodyClassName = void 0;
    var useTableLayoutStyles = react_1.makeStyles({
        root: {
            display: 'table-row-group',
        },
    });
    var useFlexLayoutStyles = react_1.makeStyles({
        root: {
            display: 'block',
        },
    });
    exports.tableBodyClassName = 'fui-TableBody';
    exports.tableBodyClassNames = {
        root: 'fui-TableBody',
    };
    /**
     * Apply styling to the TableBody slots based on the state
     */
    var useTableBodyStyles_unstable = function (state) {
        var layoutStyles = {
            table: useTableLayoutStyles(),
            flex: useFlexLayoutStyles(),
        };
        state.root.className = react_1.mergeClasses(exports.tableBodyClassName, state.noNativeElements ? layoutStyles.flex.root : layoutStyles.table.root, state.root.className);
        return state;
    };
    exports.useTableBodyStyles_unstable = useTableBodyStyles_unstable;
});
