// @fluentui/react-text/lib-commonjs/components/presets/Caption1Strong/Caption1Strong.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Caption1Strong = exports.caption1StrongClassNames = void 0;
const react_1 = /*#__PURE__*/require("@griffel/react");
const createPreset_1 = /*#__PURE__*/require("../createPreset");
const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");
exports.caption1StrongClassNames = {
  root: 'fui-Caption1Strong'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/react_1.__styles({
  root: {
    Bahqtrf: "fk6fouc",
    Be2twd7: "fy9rknc",
    Bhrd7zp: "fl43uef",
    Bg96gwp: "fwrc4pm"
  }
}, {
  d: [".fk6fouc{font-family:var(--fontFamilyBase);}", ".fy9rknc{font-size:var(--fontSizeBase200);}", ".fl43uef{font-weight:var(--fontWeightSemibold);}", ".fwrc4pm{line-height:var(--lineHeightBase200);}"]
});
/**
 * Text preset component for the Caption1Strong typography variant
 */
exports.Caption1Strong = /*#__PURE__*/createPreset_1.createPreset({
  useStyles,
  className: exports.caption1StrongClassNames.root,
  displayName: 'Caption1Strong'
});
