// @fluentui/react-text/lib-commonjs/components/presets/Caption2Strong/Caption2Strong.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Caption2Strong = exports.caption2StrongClassNames = void 0;
const react_1 = /*#__PURE__*/require("@griffel/react");
const createPreset_1 = /*#__PURE__*/require("../createPreset");
const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");
exports.caption2StrongClassNames = {
  root: 'fui-Caption2Strong'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/react_1.__styles({
  root: {
    Bahqtrf: "fk6fouc",
    Be2twd7: "f13mqy1h",
    Bhrd7zp: "fl43uef",
    Bg96gwp: "fcpl73t"
  }
}, {
  d: [".fk6fouc{font-family:var(--fontFamilyBase);}", ".f13mqy1h{font-size:var(--fontSizeBase100);}", ".fl43uef{font-weight:var(--fontWeightSemibold);}", ".fcpl73t{line-height:var(--lineHeightBase100);}"]
});
/**
 * Text preset component for the Caption2Strong typography variant
 */
exports.Caption2Strong = /*#__PURE__*/createPreset_1.createPreset({
  useStyles,
  className: exports.caption2StrongClassNames.root,
  displayName: 'Caption2Strong'
});
