// @fluentui/react-provider/lib-amd/components/FluentProvider/useFluentProviderContextValues.js
define(["require", "exports", "react"], function (require, exports, React) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useFluentProviderContextValues_unstable = void 0;
    function useFluentProviderContextValues_unstable(state) {
        // eslint-disable-next-line @typescript-eslint/naming-convention
        var applyStylesToPortals = state.applyStylesToPortals, dir = state.dir, root = state.root, targetDocument = state.targetDocument, theme = state.theme, themeClassName = state.themeClassName, overrides_unstable = state.overrides_unstable;
        var provider = React.useMemo(function () { return ({ dir: dir, targetDocument: targetDocument }); }, [dir, targetDocument]);
        // "Tooltip" component mutates an object in this context, instance should be stable
        var tooltip = React.useState(function () { return ({}); })[0];
        return {
            // eslint-disable-next-line @typescript-eslint/naming-convention
            overrides_unstable: overrides_unstable,
            provider: provider,
            textDirection: dir,
            tooltip: tooltip,
            theme: theme,
            themeClassName: applyStylesToPortals ? root.className : themeClassName,
        };
    }
    exports.useFluentProviderContextValues_unstable = useFluentProviderContextValues_unstable;
});
