// @fluentui/react-toolbar/lib-amd/components/Toolbar/useToolbarStyles.js
define(["require", "exports", "tslib", "@griffel/react"], function (require, exports, tslib_1, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useToolbarStyles_unstable = exports.toolbarClassNames = void 0;
    exports.toolbarClassNames = {
        root: 'fui-Toolbar',
    };
    /**
     * Styles for the root slot
     */
    var useStyles = react_1.makeStyles({
        root: tslib_1.__assign(tslib_1.__assign({ display: 'flex', alignItems: 'center' }, react_1.shorthands.padding('4px', '8px', '4px', '8px')), react_1.shorthands.gap('8px')),
        vertical: {
            flexDirection: 'column',
            width: 'fit-content',
        },
        small: {
            height: '32px',
        },
        medium: {
            height: '40px',
        },
        large: {
            height: '48px',
        },
    });
    /**
     * Apply styling to the Toolbar slots based on the state
     */
    var useToolbarStyles_unstable = function (state) {
        var styles = useStyles();
        var vertical = state.vertical, size = state.size;
        state.root.className = react_1.mergeClasses(exports.toolbarClassNames.root, styles.root, vertical && styles.vertical, size === 'small' && !vertical && styles.small, size === 'medium' && !vertical && styles.medium, size === 'large' && !vertical && styles.large, state.root.className);
        return state;
    };
    exports.useToolbarStyles_unstable = useToolbarStyles_unstable;
});
