// @fluentui/react-table/lib-amd/components/TableCellLayout/useTableCellLayout.js
define(["require", "exports", "tslib", "@fluentui/react-utilities", "../../contexts/tableContext"], function (require, exports, tslib_1, react_utilities_1, tableContext_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useTableCellLayout_unstable = void 0;
    var tableAvatarSizeMap = {
        medium: 32,
        small: 24,
        'extra-small': 20,
    };
    /**
     * Create the state required to render TableCellLayout.
     *
     * The returned state can be modified with hooks such as useTableCellLayoutStyles_unstable,
     * before being passed to renderTableCellLayout_unstable.
     *
     * @param props - props from this instance of TableCellLayout
     * @param ref - reference to root HTMLElement of TableCellLayout
     */
    var useTableCellLayout_unstable = function (props, ref) {
        var size = tableContext_1.useTableContext().size;
        return {
            components: {
                root: 'div',
                main: 'span',
                description: 'span',
                content: 'div',
                media: 'span',
            },
            root: react_utilities_1.getNativeElementProps('div', tslib_1.__assign({ ref: ref }, props)),
            appearance: props.appearance,
            truncate: props.truncate,
            main: react_utilities_1.resolveShorthand(props.main, { required: true }),
            media: react_utilities_1.resolveShorthand(props.media),
            description: react_utilities_1.resolveShorthand(props.description),
            content: react_utilities_1.resolveShorthand(props.content, { required: !!props.description || !!props.children }),
            avatarSize: tableAvatarSizeMap[size],
            size: size,
        };
    };
    exports.useTableCellLayout_unstable = useTableCellLayout_unstable;
});
