// @fluentui/react-virtualizer/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useIntersectionObserver = exports.useVirtualizerStyles_unstable = exports.renderVirtualizer_unstable = exports.useVirtualizer_unstable = exports.virtualizerClassNames = exports.Virtualizer = void 0;
var Virtualizer_1 = /*#__PURE__*/require("./Virtualizer");
Object.defineProperty(exports, "Virtualizer", {
  enumerable: true,
  get: function () {
    return Virtualizer_1.Virtualizer;
  }
});
Object.defineProperty(exports, "virtualizerClassNames", {
  enumerable: true,
  get: function () {
    return Virtualizer_1.virtualizerClassNames;
  }
});
Object.defineProperty(exports, "useVirtualizer_unstable", {
  enumerable: true,
  get: function () {
    return Virtualizer_1.useVirtualizer_unstable;
  }
});
Object.defineProperty(exports, "renderVirtualizer_unstable", {
  enumerable: true,
  get: function () {
    return Virtualizer_1.renderVirtualizer_unstable;
  }
});
Object.defineProperty(exports, "useVirtualizerStyles_unstable", {
  enumerable: true,
  get: function () {
    return Virtualizer_1.useVirtualizerStyles_unstable;
  }
});
var Hooks_1 = /*#__PURE__*/require("./Hooks");
Object.defineProperty(exports, "useIntersectionObserver", {
  enumerable: true,
  get: function () {
    return Hooks_1.useIntersectionObserver;
  }
});
