// @fluentui/react-text/lib-commonjs/components/presets/Subtitle2Stronger/Subtitle2Stronger.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Subtitle2Stronger = exports.subtitle2StrongerClassNames = void 0;
const react_1 = /*#__PURE__*/require("@griffel/react");
const createPreset_1 = /*#__PURE__*/require("../createPreset");
const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");
exports.subtitle2StrongerClassNames = {
  root: 'fui-Subtitle2Stronger'
};
/**
 * Styles for the root slot
 */
const useStyles = /*#__PURE__*/react_1.__styles({
  root: {
    Bahqtrf: "fk6fouc",
    Be2twd7: "fod5ikn",
    Bhrd7zp: "flh3ekv",
    Bg96gwp: "faaz57k"
  }
}, {
  d: [".fk6fouc{font-family:var(--fontFamilyBase);}", ".fod5ikn{font-size:var(--fontSizeBase400);}", ".flh3ekv{font-weight:var(--fontWeightBold);}", ".faaz57k{line-height:var(--lineHeightBase400);}"]
});
/**
 * Text preset component for the Subtitle2Stronger typography variant
 */
exports.Subtitle2Stronger = /*#__PURE__*/createPreset_1.createPreset({
  useStyles,
  className: exports.subtitle2StrongerClassNames.root,
  displayName: 'Subtitle2Stronger'
});
