// @fluentui/react-progress/lib-amd/components/ProgressBar/useProgressBar.js
define(["require", "exports", "tslib", "@fluentui/react-utilities"], function (require, exports, tslib_1, react_utilities_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useProgressBar_unstable = void 0;
    /**
     * Create the state required to render ProgressBar.
     *
     * The returned state can be modified with hooks such as useProgressBarStyles_unstable,
     * before being passed to renderProgressBar_unstable.
     *
     * @param props - props from this instance of ProgressBar
     * @param ref - reference to root HTMLElement of ProgressBar
     */
    var useProgressBar_unstable = function (props, ref) {
        // Props
        var _a = props.color, color = _a === void 0 ? 'brand' : _a, _b = props.max, max = _b === void 0 ? 1.0 : _b, _c = props.shape, shape = _c === void 0 ? 'rounded' : _c, _d = props.thickness, thickness = _d === void 0 ? 'medium' : _d, value = props.value;
        var root = react_utilities_1.getNativeElementProps('div', tslib_1.__assign({ ref: ref, role: 'progressbar', 'aria-valuemin': value !== undefined ? 0 : undefined, 'aria-valuemax': value !== undefined ? max : undefined, 'aria-valuenow': value }, props));
        var bar = react_utilities_1.resolveShorthand(props.bar, {
            required: true,
        });
        var state = {
            color: color,
            max: max,
            shape: shape,
            thickness: thickness,
            value: value,
            components: {
                root: 'div',
                bar: 'div',
            },
            root: root,
            bar: bar,
        };
        return state;
    };
    exports.useProgressBar_unstable = useProgressBar_unstable;
});
